Sources/SwiftSyntax/SyntaxData.swift

```diff
@@ -184,71 +184,78 @@ struct AbsoluteRawSyntax {
   }
 }
 
-/// Indirect wrapper for a `Syntax` node to avoid cyclic inclusion of the 
-/// `Syntax` struct in `SyntaxData`
-class SyntaxBox: CustomStringConvertible, 
-    CustomDebugStringConvertible, TextOutputStreamable {
-  let value: Syntax
+/// SyntaxData is the underlying storage for each Syntax node.
+///
+/// SyntaxData is an implementation detail, and should not be exposed to clients
+/// of SwiftSyntax.
+struct SyntaxData {
+  private enum Info {
+    case root(Root)
+    indirect case edge(Edge)
+
+    // For root node.
+    struct Root {
+      var arena: SyntaxArena
+    }
 
-  init(_ value: Syntax) {
-    self.value = value
+    // For non-root nodes.
+    struct Edge {
```

Comment:
I don’t like the term `edge` here. To me the edge would be information that’s related to the parent-child relationship between two nodes. But `absoluteInfo` is information about this node in the entire tree.

It’s not a great name but `NonRoot` would make me stumble a lot less.